app/services/search_service.rb

```diff
@@ -27,15 +27,14 @@ class SearchService
   class EmptyQuery < StandardError
   end
 
-  attr_accessor :q
-  attr_reader :result, :as_of, :data_serializer
+  attr_reader :q, :result, :as_of, :data_serializer
```

Comment:
attr_accessor :q
create also the accessor writer `q=`
which is redundant since the method `q=` already exist.